firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatController.java

```diff
@@ -118,12 +113,12 @@ private DefaultHeartBeatController(
       Context context,
       String persistenceKey,
       Set<HeartBeatConsumer> consumers,
-      Provider<UserAgentPublisher> userAgentProvider) {
+      Provider<UserAgentPublisher> userAgentProvider,
+      Executor backgroundExecutor) {
     this(
         () -> new HeartBeatInfoStorage(context, persistenceKey),
         consumers,
-        new ThreadPoolExecutor(
-            0, 1, 30, TimeUnit.SECONDS, new LinkedBlockingQueue<>(), THREAD_FACTORY),
+        backgroundExecutor,
```

Comment:
@VinayGuthal is it safe to switch to a multi-threaded executor, looks like this used to rely on a single-threaded one.

Reply:
Discussed offline, it should be safe